### PR TITLE
Fix: button component

### LINKED ISSUE #38: Text component issues
- missing style types when apply style with react native style props
- need to set new default font (opensans)
- need to check fontWeight with number(800) and text(bold) usage

### PR DESCRIPTION
**Pull request by irxd**
## What's in this MR?

- [x] Change button component usage using child
- [x] Add more stories for button component
- [x] Remove icon props
- [x] Remove title props 
- [x] Add text indicator on loading state 
- [x] Adjust button padding based on zeroheigt
- [x] Add size xs 
- [x] Add custom variant 

Screen Shot 2022-06-29 at 11 09 40
Screen Shot 2022-06-29 at 11 09 59
Screen Shot 2022-06-29 at 11 10 12
Screen Shot 2022-06-29 at 11 17 51
Screen Shot 2022-06-29 at 11 10 45

This PR resolve #38

### CODE REVIEW

**Review by andhikaribrahim — APPROVED** [review]
LGTM

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -120,6 +106,23 @@ const Button = ({
...
+        case 'xl':
+          return 18;
+        case 'lg':
+          return 16;
+        case 'md':
+          return 14;
+        case 'sm':
+          return 12;
+        case 'xs':
+          return 11;
```
Comment (nelvson) [inline src/components/Button/index.tsx:120]: `xs` sendirian engga kelipatan 2?

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -27,6 +25,7 @@ const { ButtonVariants, ButtonSizes } = Variants;
 
 export type ButtonProps = {
   variant:
+    | 'custom'
```
Comment (nelvson) [inline src/components/Button/index.tsx:28]: kalo variantnya dibikin opsional, terus default value stylingnya dari theme custom gimana?

Comment (irxd) [reply src/components/Button/index.tsx:28]: secara usage, variant udah kaya optional sih karena udah ada default propsnya

default nya set ke primary

**Review by nelvson — APPROVED** [review]

**Pull request merged**